config_converter/config_mapper/config_mapper.py

```diff
@@ -76,34 +76,39 @@ def extract_root_dirs(config_obj: config_pb2.Directive) -> tuple:
     plugin_prefix_map = {'source': 'in_'}
     dir_name_map = {'source': 'sources'}
     # these dicts can be updated when more plugins are supported
-    for d in config_obj.directives:
-        if d.name not in plugin_prefix_map:
+    for directive in config_obj.directives:
+        if directive.name not in plugin_prefix_map:
             stats['entities_skipped'] += 1
-            stats['attributes_skipped'] += _get_aggregated_num_attributes(d)
+            stats['attributes_skipped'] += _get_aggregated_num_attributes(
+                directive)
+            stats['warning_logs'] += 1
             logging.warning(
                 'Skip mapping %s due to missing functionality in master agent',
-                d.name)
+                directive.name)
             continue
         try:
-            plugin_type = next(p.value for p in d.params if p.name == '@type')
+            plugin_type = next(param.value for param in directive.params
+                               if param.name == '@type')
         except StopIteration:
+            stats['error_logs'] += 1
             logging.error('Invalid configuration - missing @type param')
             sys.exit()
-        plugin_name = plugin_prefix_map[d.name] + plugin_type
+        plugin_name = plugin_prefix_map[directive.name] + plugin_type
         if plugin_name not in _SUPPORTED_PLUGINS:
             stats['entities_unrecognized'] += 1
             stats['attributes_unrecognized'] += _get_aggregated_num_attributes(
-                d)
+                directive)
+            stats['error_logs'] += 1
             logging.error('We do not know plugin %s', plugin_name)
         else:
-            plugin_dir = dir_name_map[d.name]
+            plugin_dir = dir_name_map[directive.name]
             if plugin_dir not in logs_module:
                 logs_module[plugin_dir] = []
             current_attribute_count = stats['attributes_recognized']
             logs_module[plugin_dir].append(
-                _convert_plugin(d, plugin_name, stats))  # stats are updated
+                _convert_plugin(directive, plugin_name, stats))  # update stats
```

Comment:
I think `stats are updated` is better - as this function not only update stats but also convert plugin. 

`stats are updated` is the part that users may not notice. What do you think?

Reply:
You're right - I can't fit it on this line so I've put it on the line above. Do you think that's okay?